packages/compiler/core/schema-validator.ts

```diff
@@ -3,27 +3,33 @@ import { compilerAssert } from "./diagnostics.js";
 import { NoTarget } from "./index.js";
 import { Diagnostic, SourceFile } from "./types.js";
 
-export interface SchemaValidatorOptions {
+export interface JSONSchemaValidatorOptions {
   coerceTypes?: boolean;
 }
-export class SchemaValidator<T> {
-  private ajv: Ajv;
-
-  public constructor(private schema: JSONSchemaType<T>, options: SchemaValidatorOptions = {}) {
-    this.ajv = new Ajv({
-      strict: true,
-      coerceTypes: options.coerceTypes,
-    });
-  }
 
+export interface JSONSchemaValidator {
   /**
    * Validate the config is valid
    * @param config Configuration
    * @param target @optional file for errors tracing.
    * @returns Validation
    */
-  public validate(config: unknown, target: SourceFile | typeof NoTarget): Diagnostic[] {
-    const validate = this.ajv.compile(this.schema);
+  validate(config: unknown, target: SourceFile | typeof NoTarget): Diagnostic[];
+}
+
+export function createJSONSchemaValidator<T>(
```

Comment:
What's the reasoning for using a function instead of a class here?

Reply:
Hmm, I think I wanted to move the interface to types.ts since it is referenced by core types and then felt I would follow the pattern we have for createXxx for other core interfaces. But I see I didn't actually move the interface.

Reply:
I moved it now.